carapace-server/src/main/java/org/carapaceproxy/server/certificates/DynamicCertificatesManager.java

```diff
@@ -337,26 +327,31 @@ private void certificatesLifecycle() {
                             PrivateKey key = loadOrCreateKeyPairForDomain(domain).getPrivate();
                             String chain = base64EncodeCertificateChain(certificateChain.toArray(new Certificate[0]), key);
                             cert.setChain(chain);
-                            cert.setState(AVAILABLE);
+                            cert.success(AVAILABLE);
                             LOG.log(Level.INFO, "Certificate issuing for domain: {0} SUCCEED. Certificate AVAILABLE.", domain);
                         } else if (status == Status.INVALID) {
-                            cert.setState(REQUEST_FAILED);
+                            cert.error("Order status for certificate is " + status);
                         }
                     }
-                    case REQUEST_FAILED -> { // challenge/order failed
+                    // challenge/order failed
+                    case REQUEST_FAILED -> {
                         LOG.log(Level.INFO, "Certificate issuing for domain: {0} current status is FAILED, setting status=WAITING again.", domain);
-                        cert.setState(WAITING);
+                        if (cert.getAttemptsCount() <= getConfig().getMaxAttempts()){
```

Comment:
We can do the same thing when the certificate is always UNREACHBLE.

Reply:
Makes sense